daemons/based/based_callbacks.c

```diff
@@ -1311,10 +1311,12 @@ cib_process_command(xmlNode * request, xmlNode ** reply, xmlNode ** cib_diff, gb
         }
     }
 
-    /* Always write to disk for replace ops,
-     * this also negates the need to detect ordering changes
+    /* Always write to disk for replace and upgrade ops. This also negates
+     * the need to detect ordering changes.
      */
-    if (pcmk__str_eq(PCMK__CIB_REQUEST_REPLACE, op, pcmk__str_none)) {
+    if (pcmk__str_any_of(op,
+                         PCMK__CIB_REQUEST_REPLACE, PCMK__CIB_REQUEST_UPGRADE,
+                         NULL)) {
         config_changed = TRUE;
```

Comment:
I have no idea why we'd set this if a replace failed. We're not going to activate the new CIB if there's an error. I think we can do this only on error.

Reply:
s/on error/on success/ I assume

Hmm yeah it looks like we should already get a return code from cib_perform_op() telling us whether this succeeded. It's weird that we weren't using it already. That solves the unnecessary write problem for both operations.